sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -3362,49 +3387,18 @@ case class Sentences(
   override def second: Expression = language
   override def third: Expression = country
 
-  override def eval(input: InternalRow): Any = {
-    val string = str.eval(input)
-    if (string == null) {
-      null
-    } else {
-      val languageStr = language.eval(input).asInstanceOf[UTF8String]
-      val countryStr = country.eval(input).asInstanceOf[UTF8String]
-      val locale = if (languageStr != null && countryStr != null) {
-        new Locale(languageStr.toString, countryStr.toString)
-      } else {
-        Locale.US
```

Comment:
Reading the previous code, we use `Locale.US` if one of language and country is null. Why do we change this behavior? @panbingkun

Reply:
@MaxGekk please be careful when merging behavior changes.

Reply:
I believe old behavior is counterintuitive from user perspective. An user set a `language` explicitly, let's say to Chinese or Russian, but we ignore it, and set a geographically-specific locale linked to the USA.

Reply:
all behavior changes need to be called out in the PR description, and let's also add migration guide for it.

Reply:
Okay, let me update it to the doc `sql-migration-guide.md`.

Reply:
A new following up PR has been submitted: https://github.com/apache/spark/pull/48099